### PR TITLE
Another Proposal to fix blast aoe preview.

### PR DESCRIPTION
**Pull request by ejfudd**
Another approach to fixing the aoe preview not matching the hit calculation.

This PR just changes the rendering preview and does not touch any other code.

### CODE REVIEW

File: game/input_handlers.py [hunk 1]
```diff
@@ -308,17 +309,25 @@ def on_render(self, console: tcod.Console) -> None:
...
+                radius=self.radius,
+                light_walls=False,
+                algorithm=tcod.FOV_BASIC,
+            )
+
+            aoe_tiles &= self.engine.game_map.visible
+
+            for pos, is_aoe in np.ndenumerate(aoe_tiles):
+                if is_aoe and not (pos[0] == x and pos[1] == y):
+                    console.tiles_rgb["bg"][pos[0], pos[1]] = game.color.red
```
Comment (HexDecimal) [inline game/input_handlers.py]:
```suggestion
            aoe_tiles[x, y] = False
            console.tiles_rgb["bg"][aoe_tiles] = game.color.red 
```

**Comment by ejfudd** [comment]
This commit leaves the hit calculation as is and uses actor.distance() to determine the hit.
The shape does match the preview, but the hit calculation does not take into account tile transparency.
Basically it would just be the fov calculation without any of the additional masking for the view followed by a check if the entity is in the aoe_tile array.

File: game/input_handlers.py [hunk 2]
```diff
@@ -3,6 +3,7 @@
 from typing import Callable, Optional, Tuple
 
 import tcod
+import numpy as np
```
Comment (HexDecimal) [inline game/input_handlers.py]: isort wants these in alphabetical order.
```suggestion
import numpy as np
import tcod
```

**Comment by HexDecimal** [comment]
This is at least *more accurate* than the previous version.  I can try figure out the rest when I get to this part in my refactoring.

**Review by HexDecimal — CHANGES_REQUESTED** [review]
Uploaded code needs to be consistent since I'm doing a lot of 3-way merges with this code right now.

You'll be expected to run `black` and `isort` on your code commits to pass the linters.  This can be done automatically from some IDE's such as VSCode.

**Comment by HexDecimal** [comment]
If you still want to add to this then convert it to a draft.  Otherwise I'll merge it sometime after the tests pass.

**Comment by ejfudd** [comment]
I am fine with it as-is.  It's close enough for now and is better than before.  Adding the FOV code to the hit detection would really only allow entities the ability to have cover behind walls or around corners.  Since the tutorial does not have irregular shaped rooms or barriers inside of rooms, it's an edge case.

If you want to refine it you can address it whenever you get to part 9.

**Pull request merged**